components/script/script_runtime.rs

```diff
@@ -97,6 +97,14 @@ pub trait ScriptChan: JSTraceable {
     fn clone(&self) -> Box<dyn ScriptChan + Send>;
 }
 
+/// A cloneable interface for communicating with an event loop, from the event-loop itself.
+pub trait LocalScriptChan: JSTraceable {
+    /// Send a message to the associated event loop.
+    fn send(&self, msg: CommonScriptMsg) -> Result<(), ()>;
+    /// Clone this handle.
+    fn clone(&self) -> Box<dyn LocalScriptChan>;
```

Comment:
Can this be `impl LocalScriptChan` instead?

Reply:
Just tried it, and getting `error[E0562]: `impl Trait` not allowed outside of function and inherent method return types`